### PR TITLE
Mariya's Code Review 1

### PR DESCRIPTION
**Pull request by mooncakes-dev**
<h3> Analysis of the Program </h3>

-  Was the program available in UC Github on time? **Yes, it was**.
-   Is the program easy for you, as an outsider, to understand? **Yes, it makes sense.**
-   Does the program compile? **Yes, and tests pass.**

<h3> Technical Suggestions </h3>

1.  I changed the naming conventions in the DTO class.
2. I added a reusable navigation fragment that can be used across all of the pages. If there are components that will be used across all of the pages, there should be a fragment for it. 
3. I added Bootstrap to this application to make building out UI easier. 
4. I made a simple template for the home page (as seen in the storyboard) and made a starter for the 'View Inventory' page. In the main controller class, I made sure that you can reach the page when clicked on a button. 
5. I added a new meta tag for the HTML pages that ensures that basic accessibility standards are followed. It is important to have those tags because they are part of the web standards. If a major application does not contain them and it hinders accessibility, you can be sued. 

<h3> Technical Concepts Learned </h3>

1. I learned that you cannot have a fragments folder under resources, it has to be under templates for it to be visible by templates, which seems like a weird rule. This folder will grow exponentially, especially, if you are following atom design rules. Here is a basic [overview about atomic design](https://atomicdesign.bradfrost.com/chapter-2/) and what principles it follows. Having my components and styles nested under templates goes against the fundamental principle. Most modern applications utilize atom design to be able to swiftly make changes.
2. I learned how to create reusable components without a front-end framework. It is weird to work with html pages and using a third-part (Thyme Leaf) to make it reactive, after working with other frameworks.  For example, this code
`<div th:replace="fragments/navigation :: navigation"></div>`
it seems odd that I have to have this on every single page to have a reusable navigation. When I could have an SPA with components that update based on what is called.
3. I learned that Java has some outdated logic in regards to how much set up is required to run a simple application. All of the classes and packages that have to be set uo before your application can receive/store data. For example, I have to set up a DTO, DIO, controllers and services to have a simple CRUD application. That seems like an unnecessary amount of files.

### CODE REVIEW

File: src/main/java/com/inventorysystem/inventory/VendorProfitController.java [hunk 1]
```diff
@@ -49,10 +50,16 @@ public String addProduct(Product product){
         return "main";
     }
 
-     /** GetMapping for /product endpoint
+     /** GetMapping for /view-products endpoint
      * @return all products entered.
      */
-    @GetMapping("/product")
+
+     @RequestMapping("/view-inventory")
```
Comment (mooncakes-dev) [inline src/main/java/com/inventorysystem/inventory/VendorProfitController.java:57]: Made page name more descriptive and made sure that it mapped to the appropriate page

File: src/main/java/com/inventorysystem/inventory/VendorProfitController.java [hunk 2]
```diff
@@ -49,10 +50,16 @@ public String addProduct(Product product){
...
+
+     public String viewInventory(){
+         return "view-inventory";
```
Comment (mooncakes-dev) [inline src/main/java/com/inventorysystem/inventory/VendorProfitController.java:60]: made sure that the page is actually returned after it was called

File: src/main/resources/templates/main.html [hunk 3]
```diff
@@ -2,11 +2,27 @@
 <html xmlns:th="http://www.thymeleaf.org">
 <head>
     <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
```
Comment (mooncakes-dev) [inline src/main/resources/templates/main.html:5]: Accessibility required meta tag

File: src/main/resources/templates/main.html [hunk 4]
```diff
@@ -2,11 +2,27 @@
...
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" integrity="sha384-JcKb8q3iqJ61gNV9KGb8thSsNjpSL0n8PARn9HuZOnIxN0hoP+VmmDGMN5t9UJ0Z" crossorigin="anonymous">
     <title>Vendor Profit</title>
 </head>
 <body>
-test testing.
-<p th:text="${product}"></p>
-<p th:text="${product}" />
+<div th:replace="fragments/navigation :: navigation"></div>
```
Comment (mooncakes-dev) [inline src/main/resources/templates/main.html:10]: Added a reusable navigation component

File: src/main/resources/templates/main.html [hunk 5]
```diff
@@ -2,11 +2,27 @@
...
+
+<div class="container">
+    <div class="row">
+        <div class="col-3">
+            <nav class="nav flex-column">
```
Comment (mooncakes-dev) [inline src/main/resources/templates/main.html:15]: Basic home navigation according to the storyboard

File: src/main/java/com/inventorysystem/inventory/dto/Product.java [hunk 6]
```diff
@@ -3,10 +3,10 @@
...
 public @Data
 class Product {
-    private int ProductID;
-    private String ProductName;
-    private String ProductDesc;
-    private int VendorID;
-    private double Price;
-    private int Quantity;
+    private int productID;
+    private String productName;
```
Comment (mooncakes-dev) [inline src/main/java/com/inventorysystem/inventory/dto/Product.java:7]: Naming conventions according to the Java standards

File: src/main/java/com/inventorysystem/inventory/dto/Vendor.java [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 public @Data
 class Vendor {
-    private int VendorID;
-    private String VendorName;
-    private String PhoneNumber;
+    private int vendorID;
+    private String vendorName;
```
Comment (mooncakes-dev) [inline src/main/java/com/inventorysystem/inventory/dto/Vendor.java:8]: Naming conventions according to the Java standards

File: src/main/resources/templates/fragments/navigation.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<div th:fragment="navigation">
```
Comment (mooncakes-dev) [inline src/main/resources/templates/fragments/navigation.html:1]: This line is required in order to use this component, as you call this fragment name from the page where you needed at

File: src/main/resources/templates/main.html [hunk 9]
```diff
@@ -2,11 +2,27 @@
 <html xmlns:th="http://www.thymeleaf.org">
 <head>
     <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" integrity="sha384-JcKb8q3iqJ61gNV9KGb8thSsNjpSL0n8PARn9HuZOnIxN0hoP+VmmDGMN5t9UJ0Z" crossorigin="anonymous">
```
Comment (mooncakes-dev) [inline src/main/resources/templates/main.html:6]: Bootstrap Styles

File: src/main/resources/templates/view-inventory.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (mooncakes-dev) [inline src/main/resources/templates/view-inventory.html:1]: This template is where you can add your 'view inventory' data and call all of the methods/vars available associated with it

File: src/main/java/com/inventorysystem/inventory/VendorProfitController.java [hunk 11]
```diff
@@ -49,10 +50,16 @@ public String addProduct(Product product){
         return "main";
     }
 
-     /** GetMapping for /product endpoint
+     /** GetMapping for /view-products endpoint
      * @return all products entered.
```
Comment (discospiff) [inline src/main/java/com/inventorysystem/inventory/VendorProfitController.java:54]: As long as you're in the JavaDoc, move the first line of text to a new line.  We typically don't have any text on the /** line.

File: src/main/java/com/inventorysystem/inventory/VendorProfitController.java (hunk 1 above)
Comment (discospiff) [reply src/main/java/com/inventorysystem/inventory/VendorProfitController.java:57]: Going from GetMapping to RequestMapping makes this method more ambiguous (handle all HTTP request, not just GET).  Any reason for this?

File: src/main/java/com/inventorysystem/inventory/dao/ProductDAOStub.java [hunk 12]
```diff
@@ -23,7 +23,7 @@
     @Override
     public Product addItem(Product product) throws Exception {
         Integer productID = product.getProductID();
-        allProudcts.put(productID, product);
+        allProducts.put(productID, product);
```
Comment (discospiff) [inline src/main/java/com/inventorysystem/inventory/dao/ProductDAOStub.java:26]: Good

File: src/main/java/com/inventorysystem/inventory/dto/Product.java (hunk 6 above)
Comment (discospiff) [reply src/main/java/com/inventorysystem/inventory/dto/Product.java:7]: Good.

File: src/main/resources/templates/view-inventory.html (hunk 10 above)
Comment (discospiff) [reply src/main/resources/templates/view-inventory.html:1]: This looks like the start of a new feature.  Avoid adding new features in a code review, as a good code review should merge to master, and only features that meet the team's Definition of Done should be merged to master.  If functionality is introduced and not completed, it adds to technical debt, as it's unused code.

**Comment by discospiff** [comment]
Naming convention changes (capitalization, spelling) are worth merging, as they reduce technical debt.  These came up in another code review as well, so that can be merged in lieu of this branch, if desired.

The other changes are mostly functionality: changing endpoint name, adding new features.  Adding new features in a code review can increase technical debt, so I lean against merging these changes.  Features and functionality are the domain of the group's Product Owner.

**Comment by beaslecd** [comment]
Closing pull request

**Pull request closed without merging**